### PR TITLE
Boundary violations

### LINKED ISSUE #173: Number of boundary violations
Report number of times teams went out of bounds, in addition to time spent out of bounds.

### PR DESCRIPTION
**Pull request by yogishadixit**
Report number of times team went out of bounds.

Fixes #173

### CODE REVIEW

File: server/auvsi_suas/models/fly_zone.py [hunk 1]
```diff
@@ -122,21 +124,32 @@ def out_of_bounds(cls, fly_zones, uas_telemetry_logs):
...
-                cur_log.timestamp - prev_log.timestamp).total_seconds()
-            total_time += time_diff
-
-        return total_time
+        out_of_bounds_time = 0
+        violations = 0
+        prev_event_id = -1
+        currently_in_bounds = True
+        for i in range(len(aerial_pos_list)):
+            i_in_bounds = i not in log_ids_to_process
```
Comment (pmtischler) [inline server/auvsi_suas/models/fly_zone.py]: Now that you're traversing the `aerial_pos_list`, we should change `log_ids_to_process` to `out_of_bounds_ids`, and change it from list O(N) check to set O(1).

File: server/auvsi_suas/models/mission_config.py [hunk 2]
```diff
@@ -272,11 +272,15 @@ def evaluate_teams(self, users=None):
...
+            total_out_of_bounds_time = 0
+            total_boundary_violations = 0
             for logs in uas_period_logs:
-                out_of_bounds_time += FlyZone.out_of_bounds(
+                boundary_violations, out_of_bounds_time = FlyZone.out_of_bounds(
                     self.fly_zones.all(), logs)
-            eval_data['out_of_bounds_time'] = out_of_bounds_time
+                total_out_of_bounds_time += out_of_bounds_time
+                total_boundary_violations += boundary_violations
+            eval_data['out_of_bounds_time'] = total_out_of_bounds_time
```
Comment (pmtischler) [inline server/auvsi_suas/models/mission_config.py]: out_of_bounds_sec

File: server/server/settings.py [hunk 3]
```diff
@@ -199,6 +199,10 @@
 # (1.5x safety factor, 10Hz, 4 interfaces)
 TEST_LOADTEST_INTEROP_MIN_RATE = 1.5 * 10.0 * 4
 
+# The time (in seconds) that the plane has to be out of bounds to count as a
+# violation.
+DEBOUNCE_INTERVAL = 1.0
```
Comment (pmtischler) [inline server/server/settings.py]: Let's do 10 sec.

**Comment by pmtischler** [comment]
Don't forget to write "For #x" or "Fixes #x" so it gets tracked / closes the issue.

LGTM after comments

**Comment by yogishadixit** [comment]
Made the changes you suggested @pmtischler. PTAL

File: server/server/settings.py [hunk 4]
```diff
@@ -199,6 +199,10 @@
 # (1.5x safety factor, 10Hz, 4 interfaces)
 TEST_LOADTEST_INTEROP_MIN_RATE = 1.5 * 10.0 * 4
 
+# The time (in seconds) that the plane has to be out of bounds to count as a
```
Comment (pmtischler) [inline server/server/settings.py]: The time window (in seconds) in which a plane cannot be counted as going out of bounds multiple times.
This prevents noisy input data from recording significant more violations than a human observer.

**Comment by pmtischler** [comment]
Travis is failing. LGTM after fixing it and that one comment.

**Comment by pmtischler** [comment]
You may want to rebase the branch now that the docs build breakage is fixed.

**Pull request merged**